applications/crossbar/src/modules_v2/cb_lists_v2.erl

```diff
@@ -0,0 +1,259 @@
+-spec validate(cb_context:context(), path_token(), path_token(), path_token(), path_token()) -> cb_context:context().
+validate(Context) ->
+    case cb_context:req_verb(Context) of
+        ?HTTP_GET ->
+            crossbar_doc:load_view(<<"lists/crossbar_listing_v2">>, [], Context, fun normalize_list/2);
+        ?HTTP_PUT ->
+            validate_doc('undefined', <<"list">>, Context)
+    end.
+validate(Context, ListId) ->
+    validate_req(cb_context:req_verb(Context), Context, [ListId]).
+validate(Context, ListId, ?ENTRIES) ->
+    validate_req(cb_context:req_verb(Context), Context, [ListId, ?ENTRIES]).
+validate(Context, ListId, ?ENTRIES, EntryId) ->
+    validate_req(cb_context:req_verb(Context), Context, [ListId, ?ENTRIES, EntryId]).
+validate(Context, ListId, ?ENTRIES, EntryId, ?VCARD) ->
+    validate_req(cb_context:req_verb(Context), Context, [ListId, ?ENTRIES, EntryId, ?VCARD]).
+
+-type lists_path_tokens() :: [path_token()].
+-spec validate_req(http_method(), cb_context:context(), lists_path_tokens()) -> cb_context:context().
+validate_req(?HTTP_GET, Context, [ListId]) ->
+    crossbar_doc:load(ListId, Context);
+validate_req(?HTTP_DELETE, Context, [ListId]) ->
+    crossbar_doc:load_view(<<"lists/entries">>, [{'key', ListId}], Context);
+validate_req(?HTTP_POST, Context, [ListId]) ->
+    validate_doc(ListId, <<"list">>, Context);
+validate_req(?HTTP_PATCH, Context, [ListId] = Path) ->
+    crossbar_doc:patch_and_validate(ListId, Context, fun(_Id, C) -> validate_req(?HTTP_POST, C, Path) end);
+
+validate_req(?HTTP_GET, Context, [ListId, ?ENTRIES]) ->
+    crossbar_doc:load_view(<<"lists/entries">>, [{'key', ListId}], Context);
+validate_req(?HTTP_PUT, Context, [ListId, ?ENTRIES]) ->
+    ReqData = wh_json:set_values([{<<"list_id">>, ListId}], cb_context:req_data(Context)),
+    validate_doc('undefined', <<"list_entry">>, cb_context:set_req_data(Context, ReqData));
+validate_req(?HTTP_DELETE, Context, [ListId, ?ENTRIES]) ->
+    crossbar_doc:load_view(<<"lists/entries">>, [{'key', ListId}], Context);
+
+validate_req(?HTTP_GET, Context, [_ListId, ?ENTRIES, EntryId]) ->
+    crossbar_doc:load(EntryId, Context);
+validate_req(?HTTP_DELETE, Context, [_ListId, ?ENTRIES, EntryId]) ->
+    crossbar_doc:load(EntryId, Context);
+validate_req(?HTTP_POST, Context, [_ListId, ?ENTRIES, EntryId]) ->
+    validate_doc(EntryId, <<"list_entry">>, Context);
+validate_req(?HTTP_PATCH, Context, [_ListId, ?ENTRIES, EntryId] = Path) ->
+    crossbar_doc:patch_and_validate(EntryId, Context, fun(_Id, C) -> validate_req(?HTTP_POST, C, Path) end);
+validate_req(?HTTP_GET, Context, [_ListId, ?ENTRIES, EntryId, ?VCARD]) ->
+    Context1 = crossbar_doc:load(EntryId, Context),
+    JObj = cb_context:doc(Context1),
+    JProfile = wh_json:get_value(<<"profile">>, JObj),
+    JObj1 = wh_json:merge_jobjs(JObj, JProfile),
+    JObj2 = wh_json:set_values(
+              props:filter_empty([{<<"first_name">>, wh_json:get_value(<<"firstname">>, JObj1)}
```

Comment:
Tuples should be on their own lines, with wrapping list and function parens dropped a line:

```erlang
wh_json:set_values(
    props:filter_empty(
        [{key1, value1}
         ,{key2, value2}
         ,{key3, value3}
        ]
    )
).
```